include/igl/embree/EmbreeIntersector.h

```diff
@@ -74,9 +72,10 @@ namespace igl
       //   isStatic  scene is optimized for static geometry
       // Side effects:
       //   The first time this is ever called the embree engine is initialized.
+      template <typename DerivedV, typename DerivedF>
       inline void init(
-        const std::vector<const PointMatrixType*>& V,
-        const std::vector<const FaceMatrixType*>& F,
+        const std::vector<const Eigen::MatrixBase<DerivedV>*>& V,
+        const std::vector<const Eigen::MatrixBase<DerivedF>*>& F,
```

Comment:
So `DerivedV` and `DerivedF` should always have shape `(3, 1)` (regardless of type). It's possible that the user passes in the wrong type and gets difficult-to-understand compiler errors. Let's add some `static_asserts` on the size of `DerivedV` and `DefrivedF`. Maybe also check that `DerivedF::Scalar` is an integral type (I don't think we should support anything with fewer bits than an `int32`).